cloudinit/dmi.py

```diff
@@ -68,6 +90,29 @@ def _read_dmi_syspath(key):
     return None
 
 
+def _read_kenv(key):
+    """
+    Reads dmi data from FreeBSD's kenv(1)
+    """
+    if key not in DMIDECODE_TO_KENV_MAPPING:
+        return None
+    mapped_key = DMIDECODE_TO_KENV_MAPPING[key]
+
+    LOG.debug("querying dmi data %s", mapped_key)
+
+    try:
+        cmd = ["/bin/kenv", "-q", mapped_key]
+        (result, _err) = subp.subp(cmd)
+        result = result.strip()
+        LOG.debug("kenv returned '%s' for '%s'", result, mapped_key)
+        return result
+    except (IOError, OSError) as e:
```

Comment:
I know you just copied this, but I think you can just catch the ProcessExecutionError.  The IOError/OSError is a python2 thing, and probably was not needed there anyway as subp catches all exceptions and raises its own now.


```
>>> from cloudinit import subp
>>> x = subp.ProcessExecutionError("foo")
>>> isinstance(x, OSError)
True
>>> isinstance(x, IOError)
True
>>> isinstance(x, FileNotFoundError)
False

>>> myexc = None
>>> try:
...   subp.subp("unlikely command name")
... except Exception as e:
...   myexc = e
... 
>>> myexc
ProcessExecutionError("Unexpected error while running command.\nCommand: unlikely command name\nExit code: -\nReason: [Errno 2] No such file or directory: b'unlikely command name'\nStdout: -\nStderr: -")
```

Reply:
just catch the ProcessExecutionError. If anything else slips through subp, thats wrong